core/domain/exp_domain.py

```diff
@@ -5786,6 +5792,117 @@ def to_player_dict(self) -> ExplorationPlayerDict:
         }
 
 
+class UserExplorationDataDict(TypedDict):
+    """Dictionary representing the UserExplorationDataDict object."""
+
+    exploration_id: str
+    title: str
+    category: str
+    objective: str
+    language_code: str
+    tags: List[str]
+    init_state_name: str
+    states: Dict[str, state_domain.StateDict]
+    param_specs: Dict[str, param_domain.ParamSpecDict]
+    param_changes: List[param_domain.ParamChangeDict]
+    version: int
+    auto_tts_enabled: bool
+    edits_allowed: bool
+    draft_change_list_id: int
+    rights: rights_domain.ActivityRightsDict
+    show_state_editor_tutorial_on_load: bool
+    show_state_translation_tutorial_on_load: bool
+    is_version_of_draft_valid: Optional[bool]
+    draft_changes: Optional[Dict[str, str]]
+    email_preferences: user_domain.UserExplorationPrefsDict
+    next_content_id_index: int
+    exploration_metadata: ExplorationMetadataDict
+
+
+class UserExplorationData(translation_domain.BaseTranslatableObject):
+    """Domain object for an User Exploration data."""
+
+    def __init__(
+        self,
+        exploration: Exploration,
+        states: Dict[str, state_domain.StateDict],
+        rights: rights_domain.ActivityRightsDict,
+        exploration_email_preferences: Optional[
```

Comment:
Use domain object not dict. Dict is only for sending to frontend.

Reply:
Done, updated the type for exploration_email_preferences and we don't need to update it at two places we have a create_default_prefs function we can use that to get the default object.